### ISSUE TITLE
How to pass a subprotocol for the input puglin of websocket

### ISSUE THREAD

**Issue by yoelfme** [logstash-plugins/logstash-input-websocket#10]
Basically in Javascript I can make this
```
var ws = new WebSocket('ws://localhost:8080', 'my-own-subprotocol');
```
But the configuration for the websocket only allows the following settings

|Setting|Input Type|
|---|---|
|add_field|hash|
|codec|codec|
|mode|string, one of ["server", "client"]|
|tags|array|
|type|string|
|url|string|

[More info](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-websocket.html#_synopsis_46)

**Comment by magnusbaeck** [comment]
Indeed, it's currently not possible to send a subprotocol name along with the WebSocket handshake request. This is a reasonable feature request that AFAICT doesn't require any changes from the [ruby-ftw](https://github.com/jordansissel/ruby-ftw) WebSocket library that's used. I can't give any ETA though.

**Comment by yoelfme** [comment]
@magnusbaeck that's good :)
